command/agent/variable_endpoint.go

```diff
@@ -52,6 +72,57 @@ func (s *HTTPServer) VariableSpecificRequest(resp http.ResponseWriter, req *http
 	}
 }
 
+func (s *HTTPServer) variableLockOperation(resp http.ResponseWriter, req *http.Request,
+	operation url.Values) (interface{}, error) {
+
+	// Parse the Variable
+	var Variable structs.VariableDecrypted
+	if err := decodeBody(req, &Variable); err != nil {
+		return nil, CodedError(http.StatusBadRequest, err.Error())
+	}
+
+	if operation[renewLockQueryParam][0] == renewLockQueryParam {
```

Comment:
Don't we already know `renewLockQueryParam` is set because otherwise we wouldn't have called this method? And if not, I think this could panic on deindex?